TestFoundation/TestURL.swift

```diff
@@ -615,7 +616,15 @@ class TestURLComponents : XCTestCase {
         /* Assert that NSURLComponents.copy is actually a copy of NSURLComponents */ 
         XCTAssertTrue(copy.isEqual(urlComponent))
     }
-    
+
+    func test_hash() {
+        let c1 = URLComponents(string: "https://www.swift.org/path/to/file.html?id=name")!
```

Comment:
Same as in the other review, can we test changing all parts that contribute to the hash?